cartography/intel/github/teams.py

```diff
@@ -70,40 +82,27 @@ def _get_team_repos_for_multiple_teams(
             result[team_name] = []
             continue
 
-        repo_urls = []
-        repo_permissions = []
+        repo_urls: List[str] = []
+        repo_permissions: List[str] = []
 
-        max_tries = 5
-
-        for current_try in range(1, max_tries + 1):
+        def get_teams_repos_inner_func(
+            org: str, api_url: str, token: str, team_name: str,
+            repo_urls: List[str], repo_permissions: List[str],
+        ) -> None:
+            logger.info(f"Loading team repos for {team_name}.")
             team_repos = _get_team_repos(org, api_url, token, team_name)
-
-            try:
-                # The `or []` is because `.nodes` can be None. See:
-                # https://docs.github.com/en/graphql/reference/objects#teamrepositoryconnection
-                for repo in team_repos.nodes or []:
-                    repo_urls.append(repo['url'])
-
-                # The `or []` is because `.edges` can be None.
-                for edge in team_repos.edges or []:
-                    repo_permissions.append(edge['permission'])
-                # We're done! Break out of the retry loop.
-                break
-
-            except TypeError:
-                # Handles issue #1334
-                logger.warning(
-                    f"GitHub returned None when trying to find repo or permission data for team {team_name}.",
-                    exc_info=True,
-                )
-                if current_try == max_tries:
-                    raise RuntimeError(f"GitHub returned a None repo url for team {team_name}, retries exhausted.")
-                sleep_time_seconds = current_try ** 2
-                logger.warning(
-                    f"Waiting {sleep_time_seconds} seconds before retrying to find repo/perm for team {team_name}.",
-                )
-                sleep(sleep_time_seconds)
-
+            # The `or []` is because `.nodes` can be None. See:
+            # https://docs.github.com/en/graphql/reference/objects#teamrepositoryconnection
+            for repo in team_repos.nodes or []:
+                repo_urls.append(repo['url'])
+            # The `or []` is because `.edges` can be None.
+            for edge in team_repos.edges or []:
+                repo_permissions.append(edge['permission'])
+
+        retries_with_backoff(get_teams_repos_inner_func, TypeError, 5, backoff_handler)(
+            org=org, api_url=api_url, token=token, team_name=team_name,
+            repo_urls=repo_urls, repo_permissions=repo_permissions,
```

Comment:
1. I had to wrap the inner func in another func to work through some mypy typechecks.  (Also side benefit is it allows `retries_with_backoff` to be a reusable thing.)
2. I used kwargs instead of args to make the backoff_handler able to take some info out of the args easily, for more specific logging.